File: kth-largest-element-in-a-stream/kth-largest-element-in-a-stream.py
class KthLargest:
    def __init__(self, k: int, nums: List[int]):
        self.k = k
        self.pq = []
        for x in nums:
            heapq.heappush(self.pq, x)
            if len(self.pq) > k:
                heapq.heappop(self.pq)

    def add(self, val: int) -> int:
        heapq.heappush(self.pq, val)
        if len(self.pq) > self.k:
            heapq.heappop(self.pq)
        return self.pq[0]


# largest element an array python

File: kth-largest-element-in-a-stream/kth-largest-element-in-a-stream.java
class KthLargest {
    PriorityQueue<Integer> pq = new PriorityQueue<>();
    int k;
    public KthLargest(int k, int[] nums) {
        this.k = k;
        for(int x : nums) pq.offer(x);
        while(pq.size() > k) pq.poll();
    }
    public int add(int val) {
        pq.offer(val);
        if(pq.size() > k) pq.poll();
        return pq.peek();
    }
}

// Largest element array java